Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,142 @@
-public struct SimpleCurrencyConverter {
-    public private(set) var text = "Hello, World!"
+import Foundation
 
-    public init() {
+public enum Currency: String {
+    case JPY = "JPY"
+    case USD = "USD"
+    case GBP = "GBP"
+    case EUR = "EUR"
+}
+
+public class SimpleCurrencyConverter: CurrencyConverterProtocol {
+
+    private var exchangeRatesAPIKey: String?
+    private var networkManager: ExchangeRatesNetworkManager?
+    private static var sharedInstance: CurrencyConverterProtocol?
+    
+    /// Public setup singleton instance method for a class that allows for the setting of an exchange rates API key and a URL session.
+    ///
+    /// If an exchange rates API key is provided, it is set to the class's exchangeRatesAPIKey property.
+    /// If no exchange rates API key is provided, the class attempts to retrieve an API key from a plist file using the ExchangeRatesAPIKey keyword.
+    ///
+    /// If a URL session is provided, it is used to initialize the class's network manager.
+    /// If no URL session is provided, the class's network manager is initialized with the shared URL session.
+
+    public static func shared() -> CurrencyConverterProtocol? {
+        guard let sharedInstance = DependencyContainer.shared.resolve(type: CurrencyConverterProtocol.self) else {
```

Comment:
@ts-dzmitry-kurski Is `DependencyContainer.shared` used in other classes?

Reply:
@malayli no, according to the straightforward project structure. Would be better to do something like that at **SimpleCurrencyConverter.swift**: `let dependencyContainer = DependencyContainer()` instead of DependencyContainer singleton?

Reply:
@ts-dzmitry-kurski Yes. It is better to use `let dependencyContainer = DependencyContainer()` instead of a singleton. You can make `dependencyContainer`private if it is possible.

Reply:
@malayli done.